milli/src/index.rs

```diff
@@ -457,12 +458,43 @@ impl Index {
 
     /* searchable fields */
 
-    /// Writes the searchable fields, when this list is specified, only these are indexed.
-    pub(crate) fn put_searchable_fields(
+    /// Write the user defined searchable fields and generate the real searchable fields from the specified fields ids map.
+    pub(crate) fn put_all_searchable_fields_from_fields_ids_map(
         &self,
         wtxn: &mut RwTxn,
-        fields: &[&str],
+        user_fields: &[&str],
```

Comment:
I would prefer that you directly ask for a `Vec<String>` or `Vec<S> where S: AsRef<str>` here as you `.to_vec()` it just in this function (L476).